src/pke/include/scheme/cryptocontextparams-base.h

```diff
@@ -167,14 +169,123 @@ class Params {
         SetToDefaults(scheme0);
     }
 
+    /**
+     * This Params' constructor is to be used for unittests only.
+     *
+     * @param vals - vector with override values. sequence of vals' elements must be the same as we get it from getAllParamsDataMembers()
+     * @return a vector with names of all data members of Params
+     */
+    explicit Params(const std::vector<std::string>& vals) {
```

Comment:
This is kind of a huge method for an implicit inline.  I understand that it's not for general-purpose use.  Does it make sense to move this from a header to an implementation file?

Reply:
Agree. Will change.

Reply:
i see this has been moved to the implementation file.  thank you.